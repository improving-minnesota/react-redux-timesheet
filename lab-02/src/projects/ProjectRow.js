import React from 'react';
import PropTypes from 'prop-types';

function ProjectRow({ project }) {
  return (
    <tr>
      <td>{project.name}</td>
      <td>{project.description}</td>
    </tr>
  );
}

ProjectRow.propTypes = {
  project: PropTypes.object.isRequired
};
export default ProjectRow;
