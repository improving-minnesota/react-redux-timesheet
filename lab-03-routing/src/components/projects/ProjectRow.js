import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ProjectRow extends Component {

  render() {
    return (
      <tr>
        <td>{this.props.project.name}</td>
        <td>{this.props.project.description}</td>
      </tr>
    );
  }
}

ProjectRow.propTypes = {
  project: PropTypes.object.isRequired
};

export default ProjectRow;
