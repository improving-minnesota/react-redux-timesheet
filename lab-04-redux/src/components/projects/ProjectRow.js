import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';

class ProjectRow extends Component {
  handleClick(project) {
    if (project.deleted) {
      project.deleted = false;
      this.props.actions
        .restoreProject(project)
        .then(this.props.actions.listProjects);
    } else {
      project.deleted = true;
      this.props.actions
        .removeProject(project)
        .then(this.props.actions.listProjects);
    }
  }

  render() {
    let rowClass = '';
    if (this.props.project.deleted) {
      rowClass = 'faded';
    }

    const button = (
      <Button
        onClick={() => {
          this.handleClick(this.props.project);
        }}
        bsStyle={this.props.project.deleted ? 'success' : 'danger'}
      >
        {this.props.project.deleted ? 'Restore' : 'Delete'}
      </Button>
    );

    return (
      <tr className={rowClass}>
        <td>{this.props.project.name}</td>
        <td>{this.props.project.description}</td>
        <td>{button}</td>
      </tr>
    );
  }
}

ProjectRow.propTypes = {
  project: PropTypes.object.isRequired
};

export default ProjectRow;
