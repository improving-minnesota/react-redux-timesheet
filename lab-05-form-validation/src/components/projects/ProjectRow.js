import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {Button} from 'react-bootstrap';
import { withRouter } from 'react-router';

class ProjectRow extends Component {

  handleClick(project) {
    if(project.deleted){
      project.deleted = false;
      this.props.actions.restoreProject(project).then(this.props.actions.listProjects);
    }
    else{
      project.deleted = true;
      this.props.actions.removeProject(project).then(this.props.actions.listProjects);
    }
  }

  showDetail(project) {
    if(project.deleted) {
      console.log('You cannot edit a deleted project.');
      return;
    }

    this.props.history.push('/projects/detail/' + project._id);
  }

  render() {
    let rowClass = "";
    if(this.props.project.deleted){
      rowClass = "faded";
    }

    const button = (
      <Button
        onClick={(e) => {this.handleClick(this.props.project); e.stopPropagation();}}
        bsStyle={this.props.project.deleted ? 'success' : 'danger'}
      >
        {this.props.project.deleted ? 'Restore' : 'Delete'}
      </Button>);

    return (
      <tr className={rowClass} onClick={() => {this.showDetail(this.props.project)}}>
        <td>{this.props.project.name}</td>
        <td>{this.props.project.description}</td>
        <td>{button}</td>
      </tr>
    );
  }
}

ProjectRow.defaultProps = {
  project: {}
};

ProjectRow.propTypes = {
  project: PropTypes.object.isRequired,
  history: PropTypes.object
};

export default withRouter(ProjectRow);
