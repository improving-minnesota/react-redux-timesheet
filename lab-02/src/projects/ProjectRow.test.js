import React from 'react';
import { shallow } from 'enzyme';

import ProjectRow from './ProjectRow';

describe('<ProjectRow />', () => {
  // TODO Add tests
  let wrapper;

  beforeEach(() => {
    const project = {
      name: 'Name',
      description: 'Description'
    };
    wrapper = shallow(<ProjectRow project={project} />)
  })

  it('should instantiate the Project Row Component', () => {
    expect(wrapper).toHaveLength(1);
  })

  it('should have 2 cells, render values into expected cells', () => {
    expect(wrapper.find('td')).toHaveLength(2);
    expect(wrapper.find('td').at(0).text()).toEqual('Name');
    expect(wrapper.find('td').at(1).text()).toEqual('Description');
  })
});
