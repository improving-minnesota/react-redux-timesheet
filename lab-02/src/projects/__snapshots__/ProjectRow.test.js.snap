// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectRow /> should render to match the snapshot 1`] = `
<tr>
  <td>
    Name1
  </td>
  <td>
    Description
  </td>
</tr>
`;
